src/CoreWCF.Http/src/CoreWCF/Channels/RequestDelegateHandler.cs

```diff
@@ -97,6 +98,21 @@ private WebSocketOptions CreateWebSocketOptions(HttpTransportBindingElement tbe)
 
         internal async Task HandleRequest(HttpContext context)
         {
+            if (_httpSettings.IsAuthenticationRequired)
+            {
+                string scheme = _httpSettings.AuthenticationScheme.ToString();
+                AuthenticateResult authenticateResult = await context.AuthenticateAsync(scheme);
+                if (authenticateResult.None)
+                {
+                    await context.ChallengeAsync(scheme);
+                    return;
```

Comment:
Simplify the return

Reply:
Update: Compiler issue: Cannot have keywords after `return` when the return type is `Task`